File: 0334. Increasing Triplet Subsequence/dp.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& v) {
        if(v.size() < 3)
            return false;
        int len = v.size();
        vector<int> dp(len, 1);

        for(int i = 1; i < len; ++i){
            for(int j = 0; j < i; ++j){
                if(v[i] > v[j]){
                    dp[i] = max(dp[i], dp[j] + 1);
                }
            }
            if(dp[i] >= 3)
                return true;
        }
        return false;
    }
};

File: 0198. House Robber/dp.scala
object Solution {
    def rob(v: Array[Int]): Int = {
        var len = v.length;
        if(len == 1)
            return v(0);
        else if(len == 2)
            return math.max(v(0), v(1));
        v(1) = math.max(v(0), v(1));
        for(i <- 2 until len){
            if(v(i - 1) > v(i - 2) + v(i))
                v(i) = v(i - 1);
            else
                v(i) = v(i - 2) + v(i);
        }
        return v(len - 1);
    }
}